### PR TITLE
Add one-liner ps1 script to start Docker stack on Windows

### PR DESCRIPTION
**Pull request by johnmccabe**
Signed-off-by: John McCabe <john@johnmccabe.net>

## Description
- Adds a one-liner `deploy_stack.ps1` PowerShell script in order to start the FaaS Docker stack on Windows.
- Updates README.md to call out scripts explicitly

## Motivation and Context
When standing up the environment on Windows I defaulted to trying `docker-compose up`, only realising something was wrong when I read the warnings about `Compose does not support 'deploy' configuration`.

It also brings the user experience on Windows into line with Linux/OSX.

## How Has This Been Tested?
Have run locally on Windows 10 Pro (`10.0.15063 Build 15063`), tested the following scenarios:

- Docker not installed, returns a message pointing the user towards installing Docker.
- Swarm not initialised, the docker stack command returns a clear warning telling the user to swarm init.
- Successful deploy on a running swarm (1-node).

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/alexellis/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,12 +111,24 @@ This is my original blog post on FaaS from Janurary: [Functions as a Service blo
...
+You can quickly start FaaS on Docker Swarm online using the community-run Docker playground: play-with-docker.com (PWD) by clicking the button below:
 
 [![Try in PWD](https://cdn.rawgit.com/play-with-docker/stacks/cff22438/assets/images/button.png)](http://play-with-docker.com?stack=https://raw.githubusercontent.com/alexellis/faas/master/docker-compose.yml&stack_name=func)
 
+**Docker Swarm**
+
+A set of one-line scripts are provided to help you quickly test-drive FaaS on Docker Swarm with a set of sample functions as defined in the provided [docker-compose.yml](https://github.com/alexellis/faas/blob/master/docker-compose.yml) file.
+
+- `deploy_stack.sh` - for OSX/Linux on x86_64
+- `deploy_stack.armhf.sh` - for Linux on ARM (Raspberry Pi for example)
```
Comment (alexellis) [inline README.md]: Please can you take out the RPi reference and run this into one line?

File: deploy_stack.ps1 [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!ps1
+
+if (Get-Command docker -errorAction SilentlyContinue)
+{
+    Write-Host "Deploying stack"
+    docker stack deploy func --compose-file ./docker-compose.yml
+}
+else
```
Comment (alexellis) [inline deploy_stack.ps1]: If adding error handling for Windows - should that also be done in the Linux versions?

File: README.md [hunk 3]
```diff
@@ -111,12 +111,24 @@ This is my original blog post on FaaS from Janurary: [Functions as a Service blo
...
+- `deploy_stack.ps1` - for Windows
+
+**Kubernetes**
```
Comment (alexellis) [inline README.md]: Good idea to break this out.

File: deploy_stack.ps1 (hunk 2 above)
Comment (johnmccabe) [inline deploy_stack.ps1]: Yeah good point, I can add update the bash scripts as well, would be tonight tho.

File: README.md (hunk 1 above)
Comment (johnmccabe) [inline README.md]: don't quite get the "run this into one line" comment

Comment (alexellis) [inline README.md]: Instead of 3 bullets please can this be a single line with the two options mentioned.

**Comment by alexellis** [comment]
I tend to use Docker via Git Bash on Windows - or the Ubuntu subsystem meaning bash scripts work unaltered. If using Powershell is a common use-case then let's merge this change.

**Comment by alexellis** [comment]
Did `derek` catch you out? 🤖

**Comment by johnmccabe** [comment]
Well played derek !

@alexellis I've updated the error output for the Linux scripts as suggested, and yeah personally I stick with PowerShell on Windows where possible.

**Comment by alexellis** [comment]
Merged. Thanks @johnmccabe

**Pull request merged**